Source/Extensions/NSURLQueryItem.swift

```diff
@@ -0,0 +1,41 @@
+public extension NSURLQueryItem {
+  static func initAndEncode(key: String, value: String) -> NSURLQueryItem? {
+    guard let encodedKey = encode(key),
+          let encodedValue = encode(value)
+      else { return .None }
+
+    return NSURLQueryItem(name: encodedKey, value: encodedValue)
+  }
+
+  static func decode(data: NSData?) -> [NSURLQueryItem] {
+    guard let data = data else { return [] }
+    let result = String(data: data, encoding: NSUTF8StringEncoding)
+      .flatMap { $0.componentsSeparatedByString("&") }?
+      .map { $0.componentsSeparatedByString("=") }
+      .reduce([], combine: stringToQueryItem)
+
+    return result ?? []
+  }
+
+  private static func stringToQueryItem(accum: [NSURLQueryItem]?, el: [String]) -> [NSURLQueryItem]? {
```

Comment:
does this have to take optional `accum` and return an optional? I think this function could remove those and be fine.